.github/workflows/gitops_canary.yaml

```diff
@@ -0,0 +1,194 @@
+name: Docker Build -> Docker Hub + AWS ECR -> Gitops (canary)
+
+on:
+  workflow_call:
+    inputs:
+      name:
+        description: Friendly name of the app
+        type: string
+        required: true
+      docker-name:
+        description: Name of the docker image
+        type: string
+        required: true
+      docker-file:
+        description: Location of Dockerfile
+        type: string
+      docker-platforms:
+        description: Docker build platforms
+        type: string
+        default: linux/amd64
+      helm-chart:
+        description: Location of helm charts
+        type: string
+      gitops-repo:
+        description: Repo for gitops
+        type: string
+        required: true
+      gitops-name:
+        description: Name of the gitops app
+        type: string
+        required: true
+      gitops-values-script:
+        description: Script to run during gitops (update values.yaml)
+        type: string
+      gitops-canary-values-script:
+        description: Canary script to run during gitops (update values.yaml)
+        type: string
```

Comment:
We take another param, perhaps inputs.gitops-master-values-path to specify the location of the deployment values file to use.

We should then copy this to the canary directory, called canary.yaml or something, and then reference it in the applicationset